src/responder/common/responder_utils.c

```diff
@@ -27,6 +27,8 @@
 #include "responder/common/cache_req/cache_req.h"
 #include "util/util.h"
 
+static bool is_primary_group_request;
```

Comment:
Hi,

you cannot use global variables since SSSD can handle multiple requests in parallel. Please add this variable to the state struct of the request. Maybe you can also add an `original` to the variable name to make it clear that this is not about the primary group itself but about the special case where the original primary group is replaced. As an alternative you can also add a comment about the purpose of the variable when you declare it in the state struct.

bye,
Sumit